pkg/controller/build/build_controller.go

```diff
@@ -218,3 +244,91 @@ func (r *ReconcileBuild) validateOutputSecret(n string, ns string) error {
 	}
 	return nil
 }
+
+func (r *ReconcileBuild) configFinalizer(b *build.Build) error {
+	if b.Annotations != nil && b.Annotations[build.AnnotationBuildRunDeletion] == "true" {
+		if !contains(b.GetFinalizers(), build.BuildFinalizer) {
+			b.SetFinalizers(append(b.GetFinalizers(), build.BuildFinalizer))
+			err := r.client.Update(context.TODO(), b)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		if contains(b.GetFinalizers(), build.BuildFinalizer) {
+			b.SetFinalizers(remove(b.GetFinalizers(), build.BuildFinalizer))
+			err := r.client.Update(context.TODO(), b)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileBuild) finalizeBuildRun(b *build.Build) (bool, error) {
+	// Check if the build is marked to be deleted, which is
+	// indicated by the deletion timestamp being set.
+	if !b.DeletionTimestamp.IsZero() {
```

Comment:
is this correct to assume? I mean the deletion timestamp.

Reply:
Yes

Reply:
make sense, just learn this!